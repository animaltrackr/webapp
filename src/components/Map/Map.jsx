import React, { Component } from 'react';
import GoogleMapReact from 'google-map-react';
import Icon from 'antd/lib/icon';

class Map extends Component {
	render() {
		return (
			<div style={{ height: 'calc(100vh - 47.4px)', width: '100%' }}>
				<GoogleMapReact
					bootstrapURLKeys={{ key: process.env.REACT_APP_GOOGLE_API_KEY }}
					defaultCenter={{ lat: 48.462, lng: -123.313 }}
					defaultZoom={14}
					options={function(maps) {
						return { mapTypeId: 'satellite' };
					}}
				>
					{this.props.deerStates
						? this.props.deerStates.map((element, index) => {
								if (element['state'] || this.props.showAll) {
									return element['last_location'].map(pos => {
										console.log(pos);
										return (
											<Icon
												type="environment"
												theme="filled"
												style={{ color: element['colour'], fontSize: '18px' }}
												lat={pos[0]}
												lng={pos[1]}
												text="My Marker"
											/>
										);
									});
								}
								return '';
						  })
						: ''}
				</GoogleMapReact>
			</div>
		);
	}
}

Map.propTypes = {};

export default Map;
