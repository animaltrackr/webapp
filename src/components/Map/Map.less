.tooltip {
	& span {
		display: block;
	}
}

.pointer {
	font-size: 18px;
	cursor: pointer;
	position: relative;
	top: -20px;
	left: -10px;
}
