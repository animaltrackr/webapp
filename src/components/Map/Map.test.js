import React from 'react';
import { shallow } from 'enzyme';
import Map from './Map';

describe('<Map />', () => {
	test('renders', () => {
		const wrapper = shallow(<Map deerStates={[]} />);
		expect(wrapper).toMatchSnapshot();
	});
});
