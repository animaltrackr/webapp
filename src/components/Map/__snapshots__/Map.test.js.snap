// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> renders 1`] = `ShallowWrapper {}`;
